Importer.php

```diff
@@ -530,7 +548,14 @@ private function getRecordImporters($idSite, $viewId)
             $this->gaServiceReporting, $viewId, $this->getGoalMapping($idSite), $idSite, $quotaUser, StaticContainer::get(GoogleQueryObjectFactory::class), $this->logger);
         $gaQuery->setOnQueryMade(function () {
             ++$this->queryCount;
+            if($this->maxAvailableQueries != -1 && ($this->queryCount > $this->maxAvailableQueries)){
```

Comment:
I see 2 issues here
1. Assume you have reached the limit and when you start again next hour, 1 additional API request will still be executed even though the Quota has reached.
We may need to check at the start also if limit reached.
2. `setOnQueryMade` is being skipped for rows with no data, so assume if you have a very large date range and most of the date range has no data, we will execute more than the allowed threshold, the issue in not in this PR, but its how we have written it.
The solution to this is to move this [block of code](https://github.com/matomo-org/plugin-GoogleAnalyticsImporter/blob/4.x-dev/Google/GoogleAnalyticsQueryService.php#L122..L125) to up [here](https://github.com/matomo-org/plugin-GoogleAnalyticsImporter/blob/4.x-dev/Google/GoogleAnalyticsQueryService.php#L104)

Reply:
Hi @AltamashShaikh , 
I think issue 1 will be resolved by the PG-1519 fix. My suggestion is to keep this separate from that task.

Issue 2 : Moved the code as per your suggestion. Also, did the same to the "GoogleAnalyticsGA4QueryService.php" file as well.